config-docker.php

```diff
@@ -102,6 +102,16 @@ function getenv_docker(string $name, ?string $default = null): ?string
     'porn', 'faggot', 'sex', 'nigger', 'fuck', 'cunt', 'dick',
 ];
 
+/**
+ * include plugin configuration, if set
+ */
+$pluginconfig_dir = getenv_docker('YOURLS_PLUGIN_CONFIG_DIR', '');
+if ( $pluginconfig_dir != '' ) {
+    foreach ( glob( $pluginconfig_dir."/*.php") as $pluginconfig) {
```

Comment:
If I understand well, all the files inside a folder, for which the path is described as a volatile environment variable, are includes.

Reply:
@LeoColomb , thank you for looking into this.

define "blind"... the "admin" has to explicitly set the according config variable (YOURLS_PLUGIN_CONFIG_DIR) first, so I wouldn't call it a "blind glob inclusion".

Even in a Kubernetes environment, the files would be provisioned via i. e. a ConfigMap - and generally speaking, the security is about the same as with plug-ins themselves (the admin has to enable them, but from then on, they can do about everything triggered inside plugin.php).

Reply:
> Even in a Kubernetes environment

Keep in mind, this image can be used in different context than Kubernetes, like just a raw container.

> define "blind".

"Blind" means there are no validation, it's going to accept many risks.

While I'd be ok for one security flaw, your contribution is almost one risk per line:
* No validation on the environment
  * An attacker can change to a different path, even external
* No validation on the path
  * An misusage can quite easily break the application.
* No validation on file name
  * Literally any file on a folder can be added and loaded
  * An attacker can just dump a file in `YOURLS_PLUGIN_CONFIG_DIR`
* No validation on the PHP file
  * This validation can be tricky to manage, but still, it's open to anything, especially considering the previous points
  * It might even load a file multiple times as `include_once` is not used 

The risk here is significant, especially compared to the reasonable effort to add any config via `config.php` mount.
Does it make sense to you?

Reply:
> IOW if config-docker.php gets updated in the upstream container, I would not want to have to re-create config.php's content

Unfortunately or not, this is exactly what is expected.
Keep in mind:
* Changes in upstream `config-docker.php` are very low probable;
* In case of changes, the new version will highlight it, and with proper default if relevant;
* From maintainer perspective, it is easier to enforce user to review configuration changes "manually", than getting report of incompatible value with third party configuration.

> I might need some background info on the attack path

While YOURLS is probably not the most secure app in the world, we tend to enforce security basics on any new addition.
Security is not the assumption of safeness, but the opposite: assumption of unsafe environment. 

> unless you count in admin's faults

Exactly, but not "faults", more knowledge of the cause. If an admin wants to wildcard file inclusion, fine, but we can't hold the responsibility for every user.